src/Qir/Runtime/lib/QIR/strings.cpp

```diff
@@ -11,7 +11,7 @@
 #include "QirTypes.hpp"
 #include "QirRuntime.hpp"
 
-std::unordered_map<std::string, QirString*>& AllocatedStrings()
+std::unordered_map<std::string, QirString*>& AllocatedStrings()     // Cannot be static, is called by tests. 

```

Comment:
I'm not sure I follow why making this static blocks calling it from tests?

Reply:
The `static` function is only visible in the current translation unit (current .cpp file after preprocessing). If I make it `static`, then the linker will not be able to resolve the call to this function from the test's translation unit.

(I feel that you already know that and you mean something different. If that's the case then please elaborate)